File: 27.py
def are_ortogonal(vect_one,vect_two):
    if len(vect_one)!=len(vect_two): #primer filto: mismo tamaño
        print("los vectores deben ser del mismo tamaño")
        return None
    else:
        for element in vect_one: #comprobamos que solo contenga números
            if type(element)!=int and type(element)!=float:
                print("los vectores solo deben contener números")
                return None
        for element in vect_two:
            if type(element)!=int and type(element)!=float:
                print("los vectores solo deben contener números")
                return None
        pdto = 0
        for i in range(len(vect_one)): #para cada índice
            pdto+=vect_one[i]*vect_two[i] #vamos sumando
        if pdto == 0: #en caso que el producto punto sea nulo
            return True #si lo son
        else:
            return False #en caso contario no
are_ortogonal([1,2],[1,4,2]) #los vectores deben tener el mismo tamaño
are_ortogonal([1,2],[1,"r"])#los vectores solo deben contener números
print(are_ortogonal([1,1],[1,-1])) #son vectores ortogonales
print(are_ortogonal([1,1],[4,3])) #no son ortogonales

File: 27.js
const areOrthogonal = (vec1,vec2)=>{
    if(!Array.isArray(vec1)||!Array.isArray(vec2)||vec1.length!=vec2.length){
        return "Error, debes introducir dos array del mismo tamaño";
    }
    let pdt=0;
    for(let i=0;i<vec1.length;i++){
        try{
            vec1[i]=parseFloat(vec1[i]);
            vec2[i]=parseFloat(vec2[i]);
        }catch{
            return "Error, los vectores deben contener números";
        }
        pdt+=vec1[i]*vec2[i];
    }
    if(pdt==0) return true;
    return false;
}